infra/modules/azure_postgres_server/main.tf

```
@@ -0,0 +1,86 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.116.0"
```

```suggestion
      version = "~> 3.116.0"
```

this avoids updates to 4.x